Assets/HoloToolkit/Input/Scripts/KeywordManager.cs

```diff
@@ -84,26 +102,30 @@ void OnDestroy()
             }
         }
 
+        /// <summary>
+        /// Listens for key presses and calls the corrisponding keyword event
+        /// </summary>
         private void ProcessKeyBindings()
         {
-            foreach (var kvp in KeywordsAndResponses)
+            for (int i = 0; i < KeywordsAndResponses.Length; i++)
             {
-                if (Input.GetKeyDown(kvp.KeyCode))
+                if (Input.GetKeyDown(KeywordsAndResponses[i].KeyCode) && OnKeywordRecognized != null)
                 {
-                    kvp.Response.Invoke();
+                    OnKeywordRecognized(KeywordsAndResponses[i].Keyword);
                     return;
                 }
             }
         }
 
+        /// <summary>
+        /// Throws the OnKeywordRecognized event if the keyword is registered in the responses list.
+        /// </summary>
+        /// <param name="args">The recognized keyword.</param>
         private void KeywordRecognizer_OnPhraseRecognized(PhraseRecognizedEventArgs args)
         {
-            UnityEvent keywordResponse;
-
-            // Check to make sure the recognized keyword exists in the methods dictionary, then invoke the corresponding method.
-            if (responses.TryGetValue(args.text, out keywordResponse))
+            if (OnKeywordRecognized != null && responses.Contains(args.text))
             {
-                keywordResponse.Invoke();
+                OnKeywordRecognized(args.text);
```

Comment:
Agreed on the long answer haha! Since Unity gives us a simple way to do what we want with UnityEvents, my recommendation would be to leave this implementation as is, especially since these events only fire once when a keyword is recognized (instead of every frame or even every couple frames, when a performance increase might be more noticeable).